### PR TITLE
Stubs for Display Contract

### PR DESCRIPTION
**Pull request by thedetukov**
Implement class DisplayServiceStub, class DisplayDeviceStub

### CODE REVIEW

File: lib/src/tools/display/display_service.dart [hunk 1]
```diff
@@ -1,28 +1,49 @@
...
+  void setBrightness(DisplayDevice display, double brightness) {
+    if (brightness < 0 || brightness > 100) {
+      throw ArgumentError.value(brightness);
+    }
   }
+
+  // this.brightness = value;
+  // // ignore: avoid_print
+  // print("Set Display Brightness to $value");
+  // }
```
Comment (theanurin) [inline lib/src/tools/display/display_service.dart:25]: Remove not relevant code

File: lib/src/tools/display/display_service.dart [hunk 2]
```diff
@@ -1,28 +1,49 @@
...
 
   @override
-  void setBrightness(int value) {
-    if (value < 0 || value > 100) {
-      throw ArgumentError.value(value);
-    }
+  final bool isNightModeEnabled;
 
-    this.brightness = value;
+  // DisplayServiceStub() : this.brightness = 50;
```
Comment (theanurin) [inline lib/src/tools/display/display_service.dart:13]: Remove not relevant code

**Review by theanurin — CHANGES_REQUESTED** [review]
Why there are no any changes in controller?
I think, the project compilation is broken right now (broken compilation cannot be merged into master)

File: lib/src/tools/display/display_service.dart (hunk 1 above)
Comment (thedetukov) [reply lib/src/tools/display/display_service.dart]: Removed not relevant code

File: lib/src/tools/display/display_service.dart (hunk 2 above)
Comment (thedetukov) [reply lib/src/tools/display/display_service.dart]: Removed not relevant code

File: lib/src/tools/display/display_service.dart [hunk 3]
```diff
@@ -1,28 +1,42 @@
...
+  bool isEnabled;
 
-    // ignore: avoid_print
-    print("Set Display Brightness to $value");
-  }
+  DisplayServiceStub()
+      : this.isEnabled = true,
+        this.displays = List<DisplayDevice>.unmodifiable(
+          <DisplayDevice>[
+            DisplayServiceStub("Display_1", 30),
```
Comment (theanurin) [inline lib/src/tools/display/display_service.dart:27]: I think we want to create devices `DisplayDeviceStub` instead services `DisplayServiceStub`

```suggestion
            DisplayDeviceStub("Display_1", 30),
```

**Review by theanurin — CHANGES_REQUESTED** [review]

File: lib/src/tools/display/display_service.dart [hunk 4]
```diff
@@ -1,28 +1,42 @@
...
-      throw ArgumentError.value(value);
+  void setBrightness(DisplayDevice display, double brightness) {
+    if (brightness < 0 || brightness > 100) {
+      throw ArgumentError.value(brightness);
     }
+  }
 
-    this.brightness = value;
+  @override
+  bool isEnabled;
```
Comment (theanurin) [inline lib/src/tools/display/display_service.dart:21]: Why do you declare property that never used?

File: lib/src/tools/display/display_service.dart [hunk 5]
```diff
@@ -1,28 +1,42 @@
...
-}
+import "display_service_contract.dart" show DisplayService, DisplayDevice;
 
 class DisplayServiceStub extends DisplayService {
   @override
-  int brightness;
+  final List<DisplayDevice> displays;
+
+  @override
+  final bool isDarkModeEnabled;
```
Comment (theanurin) [inline lib/src/tools/display/display_service.dart:8]: Where a coment/documentation?

File: lib/src/tools/display/display_service.dart [hunk 6]
```diff
@@ -1,28 +1,42 @@
...
 
-  DisplayServiceStub() : this.brightness = 50;
+  @override
+  final bool isNightModeEnabled;
```
Comment (theanurin) [inline lib/src/tools/display/display_service.dart:11]: Where a coment/documentation?

File: lib/src/tools/display/display_service.dart [hunk 7]
```diff
@@ -1,28 +1,42 @@
...
 
   @override
-  void setBrightness(int value) {
-    if (value < 0 || value > 100) {
-      throw ArgumentError.value(value);
+  void setBrightness(DisplayDevice display, double brightness) {
```
Comment (theanurin) [inline lib/src/tools/display/display_service.dart:14]: Where a coment/documentation?

File: lib/src/tools/display/display_service.dart [hunk 8]
```diff
@@ -1,28 +1,42 @@
...
-  /// Call this method to enable Bluetooth adapter
-  ///
-  void setBrightness(int value);
-}
+import "display_service_contract.dart" show DisplayService, DisplayDevice;
 
 class DisplayServiceStub extends DisplayService {
   @override
-  int brightness;
+  final List<DisplayDevice> displays;
```
Comment (theanurin) [inline lib/src/tools/display/display_service.dart:5]: Where a coment/documentation?

File: lib/src/tools/display/display_service.dart [hunk 9]
```diff
@@ -1,28 +1,42 @@
...
+            DisplayServiceStub("Display_1", 30),
+            DisplayServiceStub("Display_2", 40),
+            DisplayServiceStub("Display_3", 50),
+          ],
+        );
+}
+
+class DisplayDeviceStub implements DisplayDevice {
+  @override
+  final String name;
```
Comment (theanurin) [inline lib/src/tools/display/display_service.dart:36]: Where a coment/documentation?

File: lib/src/tools/display/display_service.dart [hunk 10]
```diff
@@ -1,28 +1,42 @@
...
+
+  @override
+  final double brightness;
```
Comment (theanurin) [inline lib/src/tools/display/display_service.dart:39]: Where a coment/documentation?

**Review by theanurin — CHANGES_REQUESTED** [review]

File: lib/src/tools/display/display_service.dart (hunk 3 above)
Comment (thedetukov) [reply lib/src/tools/display/display_service.dart:27]: Yes, of course. I fixed it.

File: lib/src/tools/display/display_service.dart (hunk 4 above)
Comment (thedetukov) [reply lib/src/tools/display/display_service.dart:21]: Removed property.

File: lib/src/tools/display/display_service.dart (hunk 5 above)
Comment (thedetukov) [reply lib/src/tools/display/display_service.dart:8]: Added comment.

File: lib/src/tools/display/display_service.dart (hunk 6 above)
Comment (thedetukov) [reply lib/src/tools/display/display_service.dart:11]: Added comment.

File: lib/src/tools/display/display_service.dart (hunk 7 above)
Comment (thedetukov) [reply lib/src/tools/display/display_service.dart:14]: Added comment.

File: lib/src/tools/display/display_service.dart (hunk 8 above)
Comment (thedetukov) [reply lib/src/tools/display/display_service.dart:5]: Added comment.

File: lib/src/tools/display/display_service.dart (hunk 9 above)
Comment (thedetukov) [reply lib/src/tools/display/display_service.dart:36]: Added comment.

File: lib/src/tools/display/display_service.dart (hunk 10 above)
Comment (thedetukov) [reply lib/src/tools/display/display_service.dart:39]: Added a comment.

**Review by theanurin — APPROVED** [review]

**Pull request merged**